adapters/mqtt-vertx/src/main/java/org/eclipse/hono/adapter/mqtt/VertxBasedMqttProtocolAdapter.java

```diff
@@ -288,78 +288,74 @@ private void handleEndpointConnectionWithAuthentication(final MqttEndpoint endpo
 
     void handleCredentialsResult(final AsyncResult<String> attempt, final MqttEndpoint endpoint, final UsernamePasswordCredentials authObject) {
         if (attempt.succeeded()) {
-            String logicalDeviceId = attempt.result();
-            LOG.debug("successfully authenticated device id [{}]", logicalDeviceId);
+            final String deviceId = attempt.result();
+            LOG.debug("successfully authenticated deviceId: {}", deviceId);
             endpoint.accept(false);
 
             endpoint.publishHandler(message -> {
                 final ResourceIdentifier resource = ResourceIdentifier.fromString(message.topicName());
 
-                if (!validateCredentialsWithTopicStructure(resource, authObject.getTenantId(), logicalDeviceId)) {
+                // check that the device publishes to its device-id
+                if (!validateCredentialsWithTopicStructure(resource, authObject.getTenantId(), deviceId)) {
+
                     // if MQTT client does not conform to the topic structure, close the connection (MQTT has no way for errors)
                     endpoint.close();
                 } else {
-                    publishMessage(endpoint, authObject.getTenantId(), logicalDeviceId, message, resource);
+                    // the payload is ALWAYS published to the deviceId downstream
+                    publishMessage(endpoint, authObject.getTenantId(), deviceId, message, resource);
                 }
             });
 
             endpoint.closeHandler(v -> {
-                LOG.debug("connection closed with client [{}], authId [{}], deviceId [{}]",
-                        endpoint.clientIdentifier(), authObject.getAuthId(), logicalDeviceId);
+                LOG.debug("connection closed to device [tenantId: {}, authId: {}, deviceId: {}]",
+                        authObject.getTenantId(), authObject.getAuthId(), deviceId);
                 if (registrationAssertions.remove(endpoint) != null)
-                    LOG.trace("removed registration assertion for client [{}]", endpoint.clientIdentifier());
+                    LOG.trace("removed registration assertion for device [tenantId: {}, authId: {}, deviceId: {}]",
+                            authObject.getTenantId(), authObject.getAuthId(), deviceId);
             });
 
         } else {
-            LOG.debug("authentication failed for client [{}] rejected [{}]",
-                    endpoint.clientIdentifier(), MqttConnectReturnCode.CONNECTION_REFUSED_NOT_AUTHORIZED);
+            LOG.debug("authentication failed for device [tenantId: {}, authId: {} rejected: {}]",
```

Comment:
I guess the closing bracket is supposed to be after `authId: {}` in this case, right? Otherwise there is a comma missing and I would replace *rejected* with *error code* or *cause* ...